contracts/src/lunarswap-v1/LunarswapLibrary.compact

```diff
@@ -0,0 +1,428 @@
+   * in pair operations. The token with the smaller color becomes token0, and the
+   * larger becomes token1.
+   *
+   * Requirements:
+   * - The coins must have different colors
+   *
+   * @circuitInfo k=11, rows=600
+   *
+   * @param {CoinInfo} _tokenA - The first token to sort.
+   * @param {CoinInfo} _tokenB - The second token to sort.
+   *
+   * @throws {Error} "LunarswapLibrary: sortCoins() - Identical addresses" if tokens have same color.
+   *
+   * @returns {[CoinInfo, CoinInfo]} - The sorted tokens [token0, token1].
+   */
+  // Helper circuit to sort two coins
+  // TODO: I think it worth adding this as a generic utility for ContractAddress
+  export circuit sortCoins(_tokenA: CoinInfo, _tokenB: CoinInfo): [CoinInfo, CoinInfo] {
+    assert(_tokenA.color != _tokenB.color, "LunarswapLibrary: sortCoins() - Identical addresses");
+    const token0 = disclose(Bytes32_lt(_tokenA.color, _tokenB.color)) ? _tokenA : _tokenB;
+    const token1 = disclose(Bytes32_lt(_tokenA.color, _tokenB.color)) ? _tokenB : _tokenA;
+
+    return [token0, token1];
+  }
+
+  /**
+   * @title sortCoinsAndAmounts circuit
+   * @description Sorts coins and their corresponding amounts to maintain consistency.
+   * 
+   * @remarks
+   * This circuit sorts both coins and their amounts together to ensure the amounts
+   * correspond to the correct sorted tokens. This is essential for maintaining
+   * proper token ordering in pair operations.
+   *
+   * Requirements:
+   * - The coins must have different colors
+   *
+   * @circuitInfo k=11, rows=800
+   *
+   * @param {CoinInfo} _tokenA   - The first token to sort.
+   * @param {CoinInfo} _tokenB   - The second token to sort.
+   * @param {Uint<128>} _amountA - The amount corresponding to tokenA.
+   * @param {Uint<128>} _amountB - The amount corresponding to tokenB.
+   *
+   * @throws {Error} "LunarswapLibrary: sortCoinsAndAmounts() - Identical addresses" if tokens have same color.
+   *
+   * @returns {[CoinInfo, CoinInfo, Uint<128>, Uint<128>]} - The sorted tokens and amounts [token0, token1, amount0, amount1].
+   */
+  export circuit sortCoinsAndAmounts(
+                   _tokenA: CoinInfo, _tokenB: CoinInfo, _amountA: Uint<128>, _amountB: Uint<128>): [CoinInfo,
+                                                                                                     CoinInfo,
+                                                                                                     Uint<128>,
+                                                                                                     Uint<128>] {
+    assert(_tokenA.color != _tokenB.color, "LunarswapLibrary: sortCoinsAndAmounts() - Identical addresses");
+    if (disclose(Bytes32_lt(_tokenA.color, _tokenB.color))) {
+       // tokenA is token0
+       return [_tokenA, _tokenB, _amountA, _amountB];
+    } else {
+       // tokenB is token0
+       return [_tokenB, _tokenA, _amountB, _amountA];
+    }
```

Comment:
Would it be easier to just reuse `sortCoins`?

Reply:
Sure makes sense.